lib/phoenix_live_view/utils.ex

```diff
@@ -145,6 +145,20 @@ defmodule Phoenix.LiveView.Utils do
     |> drop_private([:connect_info, :connect_params, :assign_new])
   end
 
+  def normalize_layout(false, _warn_ctx), do: false
+  def normalize_layout(layout, _warn_ctx) when is_atom(layout), do: Atom.to_string(layout)
+
+  def normalize_layout(layout, warn_ctx) when is_binary(layout) do
+    root_template = Path.rootname(layout)
+
+    IO.warn(
+      "passing a string as a layout template in #{warn_ctx} is deprecated, please pass an atom, " <>
+        "such as :#{root_template} instead of \"#{root_template}.html\""
+    )
+
+    root_template
```

Comment:
I would convert to atom here, so we don't need to convert to string only to convert to atom?

```suggestion
  def normalize_layout(layout, _warn_ctx) when is_atom(layout), do: layout

  def normalize_layout(layout, warn_ctx) when is_binary(layout) do
    root_template = Path.rootname(layout)

    IO.warn(
      "passing a string as a layout template in #{warn_ctx} is deprecated, please pass an atom, " <>
        "such as :#{root_template} instead of \"#{root_template}.html\""
    )

    String.to_atom(root_template)
```

Reply:
Since we support setting the layout from mount options, string is safer to allow Phoenix.Template to later convert with to_existing_atom + fallback